git_history_of_project/git_history_of_project.md

```diff
@@ -0,0 +1,503 @@
+
+Let's see what that output looks like for Dracula:
+
+```
+$ git log -n 3
+commit 024af1d5d311c5d1a737cca3fb13d1b62c2a29ed (HEAD -> main, origin/main)
+Author: Vlad Dracula <vlad@tran.sylvan.ia>
+Date:   Wed Apr 6 13:17:26 2022 -0400
+
+    Ignore data files and the results folder
+
+commit 656471fa1f38c8b0b9a6a2fafdf3ba68610e61fa
+Author: Vlad Dracula <vlad@tran.sylvan.ia>
+Date:   Wed Apr 6 13:15:24 2022 -0400
+
+    Start files on Jupiter and Mercury
+
+commit 1e587d25f619aa0aa10fce19b44e4e71503fa41e
+Author: Vlad Dracula <vlad@tran.sylvan.ia>
+Date:   Wed Apr 6 13:13:21 2022 -0400
+
+    Add concerns about effects of Mars' moons on Wolfman
+```
+
+Use the arrow keys to scroll down if this output is longer than the window you are working in. If that is the case, you can always quit the output by pressing `Q`. Entering `git log` without the `-n` flag will give you the list of every commit, so if a project has been in progress for a while, this can be a lot of output!
+
+The first thing to notice is that your commit messages are here! This is a good reminder to write clear and concise messages because **future** you may be very grateful when trying to figure out where exactly **past** you introduced a particular issue.
+
+The second thing to notice is the structure of each entry in the log: commit, Author, Date, message.
+When you identify which commit you want to look at, the commit number is the 40 digit string of letters and numbers at the top of it after the word "commit". In Dracula's repository, the identifier for the commit in which he "add[ed] concerns about effects of Mars' moons on Wolfman" is `1e587d25f619aa0aa10fce19b44e4e71503fa41e`.
+
+<div class = "learnmore">
+**What is this number?**
+
+We are used to seeing numbers in base 10, which are made up of the ten digits $0,1,2,3,4,5,6,7,8,$ and $9$. Commit numbers are in base 16, called [hexadecimal](https://en.wikipedia.org/wiki/Hexadecimal). In this number system, we use the 10 familiar digits as well as the digits $a,b,c,d,e,$ and $f$ corresponding to 10,11,12,13,14, and 15 respectively in base 10.
+
+If you are using the repo you created in an earlier module, your commits might have different commit numbers than the commit numbers in the examples.
+
+On the other hand, if you created your `planets` repository identically to Dracula, with every file containing all the same characters including spaces and line breaks, your commit numbers will match Dracula's. This is because the commit number is created by putting the entire repository through a function called a **hash**. If you and Dracula put the same thing into this function, you will get the same commit number out. Be sure to use the commit number in YOUR log if it differs from Dracula's.
```

Comment:
Even though I'm who told you that identical commits will lead to identical hashes, I now think that's not true.  https://blog.thoughtram.io/git/2014/11/18/the-anatomy-of-a-git-commit.html suggests that the hash includes metadata which would differ.   In fact I just made two identical repos with identical commits and they differed (presumably because of timestamp).  Sorry for misleading you!

Reply:
I was primed to believe what you told me because the first time I made my own planets repo, it did have commit numbers that matched Carpetries, at least the 6-digit ones, which seems quite creepy if the metadata is also going in there. The timestamps should have differed even if the author (our friend Vlad Dracula) was the same.

It might be irrelevant for the module, but I definitely have more questions!